raft/doc.go

```diff
@@ -180,5 +181,100 @@ cannot be removed any more since the cluster cannot make progress.
 For this reason it is highly recommended to use three or more nodes in
 every cluster.
 
+MessageType
+
+Package raft sends and receives message in Protocol Buffer format (defined
+in raftpb package). Each state (follower, candidate, leader) implements its
+own 'step' method ('stepFollower', 'stepCandidate', 'stepLeader') when
+advancing with the given raftpb.Message. Each step is determined by its
+raftpb.MessageType. Note that every step is checked by one common method
+'Step' that safe-checks terms of node and incoming message to prevent stale
+log entries:
+
+	'MsgHup' is used for election. If a node is follower or candidate, the
+	tick function in 'raft' struct is set as 'tickElection'. If follower or
```

Comment:
`tick` function in the `raft` structure is set as `tickElection`. If a follower or